lib/hammer_cli_foreman/auth.rb

```diff
@@ -46,16 +46,16 @@ def execute
           if option_two_factor?
             Oauth.execute_with_params(
               AUTH_TYPES[:oauth_authentication_code_grant],
-              option_oidc_token_endpoint,
-              option_oidc_authorization_endpoint,
-              option_oidc_client_id,
-              option_oidc_redirect_uri
+              option_oidc_token_endpoint || HammerCLI::Settings.get(:foreman, :oidc_token_endpoint),
+              option_oidc_authorization_endpoint || HammerCLI::Settings.get(:foreman, :oidc_authorization_endpoint),
+              option_oidc_client_id || HammerCLI::Settings.get(:foreman, :oidc_client_id),
+              option_oidc_redirect_uri || HammerCLI::Settings.get(:foreman, :oidc_redirect_uri)
```

Comment:
I am not sure if this should be passed here. I want to make sure this change does not have any side effects. Will further test and let you know.

Reply:
BTW, a lot of the testing also depends upon this conversation: https://github.com/theforeman/hammer-cli-foreman/pull/516/files#r420009404 lets resolve this conversation and make a decision.